rest-api-json-bulk/src/main/kotlin/org/modelix/sample/restapijsonbulk/BulkApi.kt

```diff
@@ -34,16 +39,17 @@ object RouteHelper {
 fun Route.BulkApi(loadRoots: suspend () -> List<INode>, resolve: suspend (INodeReference) -> INode?) {
     get<Paths.getLectures> {
         val roots = loadRoots()
+
         val allLectures = roots.flatMap {
-            (it.allChildren.map { MPSLanguageRegistry.getInstance<BaseConcept>(it) })
-        }.filterIsInstance<Courses>().flatMap { it.children.lectures }
+            (it.allChildren.map { it.typed() })
+        }.filterIsInstance<N_Courses>().flatMap { it.lectures }
 
-        val roomList = LectureList(lectures = allLectures.map {
-            Lecture(name = it.properties.name!!,
-                    description = it.properties.description!!,
-                    lectureRef = RouteHelper.urlEncode(INodeReferenceSerializer.serialize(it.iNode.reference)),
-                    room = RouteHelper.urlEncode(INodeReferenceSerializer.serialize(it.references.room.iNode.reference)),
-                    maxParticipants = it.properties.maxParticipants!!)
+        val roomList = LectureList(lectures = allLectures.map { lectureInstance ->
+            Lecture(name = lectureInstance.name!!,
+                    description = lectureInstance.description!!,
+                    lectureRef = RouteHelper.urlEncode(lectureInstance.unwrap().reference.serialize()),
```

Comment:
What is this `unwrap` actually doing?

Reply:
as far as i know provides access to the actual wrapped INode underneath. the only way to access the reference itself without castings such as `(lectureInstance.room as TypedNodeImpl).wrappedNode.reference`